test/planwise/component/datasets2_test.clj

```diff
@@ -85,3 +113,19 @@
           listed-sites             (datasets2/sites-by-version store dataset-id last-version-dataset)]
       (is (= (count listed-sites) 2))
       (is (= last-version-dataset 2)))))
+
+;; ----------------------------------------------------------------------
+;; Testing site's tag filtering
+
+(deftest filtering-sites
+  (test-system/with-system (test-config fixture-filtering-sites-tags)
```

Comment:
This is ok, but you don't need to spin up a test system to test the filtering functionality. Since it's just a side-effect free function with no real dependencies, you can construct the sites collection and use it to invoke `planwise.component.datasets2/filter-sites-by-tags` directly